pypeit/scripts/collate_1d.py

```diff
@@ -0,0 +1,823 @@
+#!/usr/bin/env python
+#
+# See top-level LICENSE file for Copyright information
+#
+# -*- coding: utf-8 -*-
+"""
+This script collates multiple 1d spectra in multiple files by object, 
+runs flux calibration/coadding on them, and produces files suitable
+for KOA archiving.
+"""
+
+import sys
+from datetime import datetime, timezone
+import argparse
+from glob import glob
+import os.path
+import shutil
+from functools import partial
+import re
+
+import numpy as np
+from astropy.time import Time
+import astropy.units as u
+from astropy.coordinates import SkyCoord, Angle
+from astropy.io import fits, ascii
+from astropy.table import Table
+
+from pypeit import specobjs
+from pypeit.par import pypeitpar
+from pypeit.spectrographs.util import load_spectrograph
+from pypeit import coadd1d
+from pypeit.spec2dobj import AllSpec2DObj, Spec2DObj
+from pypeit import msgs
+from pypeit import par
+from pypeit.slittrace import SlitTraceBitMask
+
+# A trick from stackoverflow to allow multi-line output in the help:
+#https://stackoverflow.com/questions/3853722/python-argparse-how-to-insert-newline-in-the-help-text
+class SmartFormatter(argparse.HelpFormatter):
+
+    def _split_lines(self, text, width):
+        if text.startswith('R|'):
+            return text[2:].splitlines()
+        # this is the RawTextHelpFormatter._split_lines
+        return argparse.HelpFormatter._split_lines(self, text, width)
+
+class KOAArchiveDir():
```

Comment:
Do we really want to make the name KOA specific? We support non-Keck instruments as well, which this should also work with. We can adopt the KOA framework, but maybe name it something different.

Reply:
I agree, and changed it to ArchiveDir.